internal/controller/rabbitmq/rabbitmqpolicy_controller.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbitmq
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	rabbitmqv1 "github.com/openstack-k8s-operators/infra-operator/apis/rabbitmq/v1beta1"
+	rabbitmqapi "github.com/openstack-k8s-operators/infra-operator/pkg/rabbitmq/api"
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	helper "github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	oko_secret "github.com/openstack-k8s-operators/lib-common/modules/common/secret"
+	rabbitmqclusterv2 "github.com/rabbitmq/cluster-operator/v2/api/v1beta1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+const policyFinalizer = "rabbitmqpolicy.openstack.org/finalizer"
+
+// RabbitMQPolicyReconciler reconciles a RabbitMQPolicy object
+//
+//nolint:revive
+type RabbitMQPolicyReconciler struct {
+	client.Client
+	Kclient kubernetes.Interface
+	Scheme  *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=rabbitmqpolicies,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=rabbitmqpolicies/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=rabbitmqpolicies/finalizers,verbs=update
+
+// Reconcile reconciles a RabbitMQPolicy object
+func (r *RabbitMQPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	Log := log.FromContext(ctx)
+
+	instance := &rabbitmqv1.RabbitMQPolicy{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	h, _ := helper.NewHelper(instance, r.Client, r.Kclient, r.Scheme, Log)
+
+	// Initialize status
+	if instance.Status.Conditions == nil {
```

Comment:
I think we should do it like in the other controllers:
* save the conditions https://github.com/openstack-k8s-operators/infra-operator/blob/main/internal/controller/rabbitmq/rabbitmq_controller.go#L146
* always init the conditions https://github.com/openstack-k8s-operators/infra-operator/blob/main/internal/controller/rabbitmq/rabbitmq_controller.go#L175
* in the defer restore the timestamp if no change func https://github.com/openstack-k8s-operators/infra-operator/blob/main/internal/controller/rabbitmq/rabbitmq_controller.go#L158

with this if we add a new condition it gets added properly when updating an env.